instaloader/nodeiterator.py

```diff
@@ -157,6 +163,11 @@ def magic(self) -> str:
         ).encode())
         return base64.urlsafe_b64encode(magic_hash.digest()).decode()
 
+    @property
+    def first_item_timestamp(self) -> Optional[datetime]:
+        """If this iterator has produced any items, returns the timestamp of that item, in UTC."""
```

Comment:
This would probably be an appropriate place for the `.. versionadded:: 4.8` directive.